### DISCUSSION TITLE
[Question]: Frigate 14 beta2, live mode -  how it works?

### DISCUSSION THREAD

**Discussion by lineumaciel** [blakeblackshear/frigate#11724] [Ask A Question] [answered]
### What is your question:

Hi,

How exactly does new “smart” live mode work?. MSE is always selected?. 

Regardless of the stream configurations in go2rtc, MSE is always used and there is no way to switch to a stable stream in WebRTC.

While it works well on desktop browsers, mobile versions are a real nightmare.

Is there any chance to return to live_mode which was removed with beta2?

I'm not saying the new solution is bad but for me personally the ability to auto/mse/webrtc in live mode would be just perfect.

**Comment by NickM-27** [comment]
why does mse not work on mobile? what mobile device, browser are you on?

**Reply by lineumaciel** [reply]
I'm not saying that MSE doesn't work. But compared to WebRTC, MSE on mobile safari leaves a lot to be desired.


In my case, WebRTC performs much better than MSE. I have already changed go2rtc configurations and eliminated the two-way audio that MSE has a problem with(no picture). 


Clicking microphone to make the stream switch from MSE to WebRTC is a bit cumbersome and for me it is a step backwards.


I thought that disabling the flag in Safari would allow switching to WebRTC automatically but it results in a message about no support for MSE.


I have checked several ios 17.5 devices and everywhere i have the same symptom.


In my test configuration it was:
     
`
      go2rtc:
    
      streams:
        tapo:
          - tapo://adminadmin@192.168.55.44
        tapolive:
          - ffmpeg:tapo#video=h264#audio=aac#hardware=vaapi
        tapodetect:
          - ffmpeg:tapo#video=h264#width=1280#raw=-fpsmax 10#hardware=vaapi
    
    cameras:
    
      tapo:
        enabled: true
        ffmpeg:
          output_args:
            record: preset-record-generic
          inputs:
            - path: rtsp://localhost:8554/tapo
              input_args: preset-rtsp-restream
              roles:
                - record
            - path: rtsp://localhost:8554/tapodetect
              input_args: preset-rtsp-restream
              roles:
                - detect

    live:
      stream_name: tapo`
      
It worked without any problems until version 14 beta1. Now the stream is not showing an image. The image appears when I click on the microphone icon(stream changes from MSE to WebRTC). I currently use as live tapolive.

`        {
            "type": "MSE/WebSocket active consumer",
            "remote_addr": "192.168.4.250, ::ffff:172.18.0.2, 127.0.0.1:58822",
            "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36 Edg/125.0.0.0",
            "medias": [
                "video, sendonly, H264, H265",
                "audio, sendonly, MPEG4-GENERIC, PCMA, PCMU, L16, PCML, OPUS"
            ],
            "senders": [
                "255 H264, bytes=3674439, receivers=1",
                "8 FLAC/8000, bytes=660992, receivers=1"
            ],
            "send": 5422764
        },
        {
            "type": "MSE/WebSocket active consumer",
            "remote_addr": "192.168.4.232, ::ffff:172.18.0.2, 127.0.0.1:47742",
            "user_agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 17_5_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.5 Mobile/15E148 Safari/604.1",
            "medias": [
                "video, sendonly, H264, H265",
                "audio, sendonly, MPEG4-GENERIC"
            ],
            "senders": [
                "255 H264, bytes=451658, receivers=1"
            ],
            "send": 475221
        }
    ]
}`


![IMG_6418](https://github.com/blakeblackshear/frigate/assets/70479212/36d22081-04a0-4519-8eac-2ba9254b989a)
![IMG_6419](https://github.com/blakeblackshear/frigate/assets/70479212/f8611b1a-5e86-4cb6-aaf6-b045876410b0)

**Reply by hawkeye217** [reply, marked as answer]
It's hard to know exactly what's happening in your situation, but I have committed some fixes for a very specific issue I was seeing on one of my own cameras. Let me know if this improves it for you. https://github.com/blakeblackshear/frigate/pull/11745

**Reply by lineumaciel** [reply]
It works perfectly :-), thank you!

**Reply by hawkeye217** [reply]
Great!
